usr/src/uts/aarch64/krtld/kobj_isa.c

```diff
@@ -81,8 +81,8 @@ void
 kobj_sync_instruction_memory(caddr_t addr, size_t len)
 {
 	uint64_t ctr = read_ctr_el0();
-	uint64_t inst_line_size = CTR_IMINLINE_SIZE(ctr);
```

Comment:
I'm confused about the prefetch buffer thing.  Is the CTR_* stuff also incorrectly implemented?
Moving the bitx stuff into unix is also a thing we can do.

Reply:
The prefetch buffer is more the only way I can explain the behaviour. On bugger Arm CPUs (definitely Neoverse-V2), the prefetch buffer is ~250 instructions.

CTR_ is perfectly OK, but we've already prefetched an un-relocated bitx64 call by the time we run this code when we're on such a CPU, which results in a very early trap.

I think the reason we don't see this on rpi4 is that the prefetch buffer is so small.

I went with open-coding this rather than bringing bitx64 into unix since krtld is really supposed to be standalone, but I'm not opposed to just linking the required function in.